WordPress/src/main/java/org/wordpress/android/ui/me/MeFragment.java

```diff
@@ -101,13 +108,54 @@ private void refreshAccountDetails() {
             }
 
             mLoginLogoutTextView.setText(R.string.me_disconnect_from_wordpress_com);
-        }
-        else {
+            mLoginLogoutTextView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    signoutWithConfirmation();
+                }
+            });
+        } else {
             mAvatarImageView.setVisibility(View.GONE);
             mDisplayNameTextView.setVisibility(View.GONE);
             mUsernameTextView.setVisibility(View.GONE);
 
             mLoginLogoutTextView.setText(R.string.me_connect_to_wordpress_com);
+            mLoginLogoutTextView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    ActivityLauncher.showSignInForResult(getActivity());
+                }
+            });
         }
     }
+
+    private void signoutWithConfirmation() {
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity());
+        dialogBuilder.setMessage(getString(R.string.sign_out_confirm));
+        dialogBuilder.setPositiveButton(R.string.signout, new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int whichButton) {
+                signout();
+            }
+        });
+        dialogBuilder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int whichButton) {
```

Comment:
> If the button has no action you can just pass null for the OnClickListener

Good catch! I simply copied the existing code from Settings - I'll correct this.